DEVDOCS-1977: Beginning Development (#511)
bigcommerce/dev-docs: docs/api-docs/apps/guide/apps-02-types.md

<div class="otp" id="no-index">

### On this page

> - [Single-click](#single-click)

Reviewer: ```suggestion
- [Single-Click](#single-click)

```

- [Connector](#connector)
- [Visibility](#visibility)
- [Next steps](#next-steps)
- [Resources](#resources)
</div>

The first step when developing an app is deciding which type of app to develop. The two types of apps, single-click and connector, are defined by the method of authentication. [Single-click](#single-click) apps use an OAuth Authorization Code Grant flow. [Connector apps](#connector) require store owners to manually generate and configure store API credentials. In addition to the authentication method, apps can also differ by [visibility](#visibility).

> ## Single-click

Reviewer: ```suggestion
## Single-Click

```

<div class="otp" id="no-index">

### On This Page
- [Single Click](#single-click)
- [Connector](#connector)
- [Visibility](#visibility)
- [Next Steps](#next-steps)
- [Resources](#resources)

</div>

> The first step to developing an app is deciding which type of app to develop. The two types of apps, single-click and connector, are defined by method of authentication. [Single-click](#single-click) apps use an OAuth Authorization Code Grant flow. [Connector apps](#connector) require store owners to manually generate and configure store API credentials. In addition to authentication method, apps can also differ by [visibility](#visibility).

Author: I'm not sure what's wrong with the way I worded it.

> ## Single Click

Reviewer: ```suggestion
## Single-Click

```

Single-click apps are the recommended app type. They use [OAuth 2.0 Authorization Code Grant](https://oauth.net/2/grant-types/authorization-code/) (oauth2.net) flow. Users initiate installation by clicking **Install**:

![Install App](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-02-types-01.png "Install App")

The app requests the user to grant permissions associated to scopes configured for the app in [DevTools](https://devtools.bigcommerce.com/my/apps):

![App Permissions](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-02-types-02.png "App Permissions")

Once granted, the app can request a permanent `access_token` for making REST API requests on the user's behalf.

## Connector

> Connector apps use manual OAuth token creation instead of the single-click app flow. Store owner's generate [Store API Credentials](https://developer.bigcommerce.com/api-docs/getting-started/authentication/rest-api-authentication#revoking-store-api-credentials) manually and enter them into the app's configuration. While single-click apps are recommended, certain use cases might not be compatible:

Author: This changes the meaning from "here's some circumstances" to use connector app to "here's the **only** circumstances" to use connector apps. Are you sure those are the **only** circumstances connector apps should be used?

- Customized integrations that vary per store.
- Integrations that do not provide any content for an iFrame.
- Apps that do not provide web services for OAuth exchange (ex: native order sync app).

> ## Visibility

Reviewer: The information below would be better structured as a table.

[List or table](https://intranet.bigcommerce.com/display/SST/Tables#Tables-Listortable?)

Author: Use cases is a list that might grow over time. Its better to have in a list because tables with lots of text do not look very good.

Reviewer: Having a running list of bullet points that separate the use cases from the original visibility option also does not look good. Let me know if you need help making this into a table.

> There are three visibility options for apps: **draft**, **unlisted**, and **public**.

Reviewer: ```suggestion
There are three visibility options for apps: **Draft**, **Unlisted**, and **Public**.

```

* **Draft**
  * **Description**: apps registered in [Dev Tools](https://devtools.bigcommerce.com/) but never submitted for approval

>   * **Use Cases:**:

Reviewer: ```suggestion
  * **Use Cases:**

```

    * installing on stores owned by the same email address as [DevTools](https://devtools.bigcommerce.com/auth/bigcommerce) account email
    * beta testing before publishing to marketplace
* **Unlisted**

>   * **Description:**: apps submitted and approved apps but not listed on [app marketplace](https://www.bigcommerce.com/apps/).

Reviewer: ```suggestion
  * **Description:** apps submitted and approved apps but not listed on [app marketplace](https://www.bigcommerce.com/apps/).

```

  * **Use Cases:**
    * installing on stores not owned by same email address as [**DevTools**](https://devtools.bigcommerce.com/auth/bigcommerce) account email
    * beta testing before publishing to marketplace
* **Public**

>   * **Description**: apps submitted, approved, and listed on the marketplace

Reviewer: ```suggestion
  * **Description:** apps submitted, approved, and listed on the marketplace

```

<div class="otp" id="no-index">

> ### On This Page

Reviewer: The section headings on this page should be [sentence case](https://intranet.bigcommerce.com/display/SST/Headings+and+Titles)

- [Single-Click](#single-click)

- [Connector](#connector)
- [Visibility](#visibility)
- [Next Steps](#next-steps)
- [Resources](#resources)

</div>

The first step when developing an app is deciding which type of app to develop. The two types of apps, single-click and connector, are defined by the method of authentication. [Single-click](#single-click) apps use an OAuth Authorization Code Grant flow. [Connector apps](#connector) require store owners to manually generate and configure store API credentials. In addition to the authentication method, apps can also differ by [visibility](#visibility).

## Single-Click

> Single-click apps are the recommended app type. They use [OAuth 2.0 Authorization Code Grant](https://oauth.net/2/grant-types/authorization-code/) (oauth2.net) flow. Users initiate installation by clicking **Install**:

Reviewer: ```suggestion
Single-click apps are the recommended app type. They use [OAuth 2.0 Authorization Code Grant](https://oauth.net/2/grant-types/authorization-code/) (oauth2.net) flow. Users initiate installation by clicking **Install**.

```

![Install App](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-02-types-01.png "Install App")

The app requests the user to grant permissions associated to scopes configured for the app in [DevTools](https://devtools.bigcommerce.com/my/apps):

![App Permissions](https://raw.githubusercontent.com/bigcommerce/dev-docs/master/assets/images/apps-02-types-02.png "App Permissions")

Once granted, the app can request a permanent `access_token` for making REST API requests on the user's behalf.

## Connector

> Connector apps use manual OAuth token creation instead of the single-click app flow. Store owner's generate [Store API Credentials](https://developer.bigcommerce.com/api-docs/getting-started/authentication/rest-api-authentication#revoking-store-api-credentials) manually and enter them into the app's configuration. While single-click apps are recommended, certain use cases might not be compatible:

Reviewer: ```suggestion
Connector apps use manual OAuth token creation instead of the single-click app flow. Store owners generate [store API credentials](https://developer.bigcommerce.com/api-docs/getting-started/authentication/rest-api-authentication#revoking-store-api-credentials) manually and enter them into the app's configuration. While single-click apps are recommended, the following use cases might not be compatible:

```

- Customized integrations that vary per store.
- Integrations that do not provide any content for an iFrame.
- Apps that do not provide web services for OAuth exchange (ex: native order sync app).

## Visibility
There are three visibility options for apps: **Draft**, **Unlisted**, and **Public**.

* **Draft**
  * **Description**: apps registered in [Dev Tools](https://devtools.bigcommerce.com/) but never submitted for approval
  * **Use Cases:**

    * installing on stores owned by the same email address as [DevTools](https://devtools.bigcommerce.com/auth/bigcommerce) account email
    * beta testing before publishing to marketplace
* **Unlisted**
  * **Description:** apps submitted and approved apps but not listed on [app marketplace](https://www.bigcommerce.com/apps/).

  * **Use Cases:**
    * installing on stores not owned by same email address as [**DevTools**](https://devtools.bigcommerce.com/auth/bigcommerce) account email
    * beta testing before publishing to marketplace
* **Public**
  * **Description:** apps submitted, approved, and listed on the marketplace

  * **Use Cases:**
    * making apps visible and installable on all stores

>     * making $$$

Reviewer: Clarify this use case
